torch/csrc/jit/serialization/pickler.cpp

```diff
@@ -803,4 +803,18 @@ bool checkHasValidSetGetState(const std::shared_ptr<c10::ClassType>& cls) {
   return true;
 }
 
+std::array<
+    c10::optional<std::pair<BackendMetaPtr, BackendMetaPtr>>,
+    at::COMPILE_TIME_MAX_DEVICE_TYPES>&
+GetBackendMetaSerialization() {
```

Comment:
Can you move this back to the header file by any chance? That would solve the build issue that is going to painful to investigate haha

Reply:
Of course.

But there are some tiny differences between gcc and clang in defining global variables in inline function, which will cause the test failure if compiled by clang. Moving the function to .cc is the most easiest way to fix the problem.

So if we want to move the function back, I need sometime to dive into the problem and find out the root cause.

Alternative, I can disable the related test temporarily and re-enable it when fixed.

Reply:
Ok,
This is very weird because getWriteableTensorData method above is defined the same way as your updated one but does work...

Reply:
I can't think of the problem right now.
Let's reland the rest of the fix with skips for the tests that are indeed broken.
This will at least ensure the rest is properly tested. We can look into the fix for this independently.

Reply:
Got it.
I will submit a new PR today.